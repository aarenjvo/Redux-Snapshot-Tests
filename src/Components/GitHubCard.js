import Card from 'react-bootstrap/Card'
import Button from 'react-bootstrap/Button'
import React from 'react'
import { useEffect } from 'react'

function GitHubCard({ userData }) {
    useEffect(() => {
        console.log('display user data mounted')
    }, [])
    return (
        <div>
            <h1>{userData.login}</h1>
            <img src={userData.avatar_url} alt='profile' />
            {userData.name && <h2>Name: {userData.name}</h2>}
            <p>
                <a href={userData.html_url} target="blank">Profile</a>
            </p>
            {userData.bio && <p>Bio: {userData.bio}</p>}
            {userData.twitter_username && <p>Twitter: {userData.twitter_username}</p>}
        </div>
    );
}

export default GitHubCard