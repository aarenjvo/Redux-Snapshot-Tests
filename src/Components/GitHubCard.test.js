import renderer from 'react-test-renderer'
import GitHubCard from './GitHubCard'

test('renders a snapshot of GitHubCard', () => {
    const userData = {
        login: '',
        avatar_url: '',
        html_url: '',
        name: '',
        bio: '',
        twitter_username: ''
    }
    const tree = renderer.create(<GitHubCard userData={userData} />).toJSON()
    expect(tree).toMatchSnapshot()
})
