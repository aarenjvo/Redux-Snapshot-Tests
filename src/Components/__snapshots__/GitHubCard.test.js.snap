// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of GitHubCard 1`] = `
<div>
  <h1 />
  <img
    alt="profile"
    src=""
  />
  <p>
    <a
      href=""
      target="blank"
    >
      Profile
    </a>
  </p>
</div>
`;
